File: ckgame-java/src/main/java/com/ckgame/world/Dynasty.java
package com.ckgame.world;

import com.ckgame.core.Constants;

import java.util.ArrayList;
import java.util.List;

/** 王朝：家族实体，成员列表与家族威望。 */
public final class Dynasty {
    public int id;
    public String name;
    public int head = Constants.NONE_ID;
    public int founder = Constants.NONE_ID;
    public List<Integer> members = new ArrayList<>();
    public int colorR = 128, colorG = 128, colorB = 128;
    public String motto = "";
    public double prestige = 0.0;

    public Dynasty(int id, String name) {
        this.id = id;
        this.name = name;
    }

    public static Dynasty newDynasty(int dynastyId, String name, int head) {
        Dynasty d = new Dynasty(dynastyId, name);
        d.head = head;
        d.founder = head;
        return d;
    }

    public void addMember(int who) {
        if (!members.contains(who)) {
            members.add(who);
        }
    }
}

File: ckgame-ts/src/world/Dynasty.ts
import { NONE_ID } from '../core/Constants.js';

/** 王朝：家族实体，成员列表与家族威望。 */
export class Dynasty {
  id: number;
  name: string;
  head: number = NONE_ID;
  founder: number = NONE_ID;
  members: number[] = [];
  colorR: number = 128;
  colorG: number = 128;
  colorB: number = 128;
  motto: string = '';
  prestige: number = 0;

  constructor(id: number, name: string) {
    this.id = id;
    this.name = name;
  }

  static newDynasty(dynastyId: number, name: string, head: number): Dynasty {
    const d = new Dynasty(dynastyId, name);
    d.head = head;
    d.founder = head;
    return d;
  }

  addMember(who: number): void {
    if (!this.members.includes(who)) {
      this.members.push(who);
    }
  }
}
